src/Microsoft.TestPlatform.CrossPlatEngine/PostProcessing/ArtifactProcessingManager.cs

```diff
@@ -0,0 +1,219 @@
+        }
+
+        if (string.IsNullOrEmpty(_testSessionCorrelationId))
+        {
+            EqtTrace.Verbose("ArtifactProcessingManager.CollectArtifacts: null testSessionCorrelationId");
+            return;
+        }
+
+        try
+        {
+            // if we have some invoked data collector with an attachment processor registered we save the outcome and the runsettings
+            if (testRunCompleteEventArgs?.InvokedDataCollectors.Count(x => x.HasAttachmentProcessor) > 0)
+            {
+                EqtTrace.Verbose($"ArtifactProcessingManager.CollectArtifacts: Saving data collectors artifacts for post process into {_processArtifactFolder}");
+                Stopwatch watch = Stopwatch.StartNew();
+                _fileHelper.CreateDirectory(_testSessionProcessArtifactFolder);
+                EqtTrace.Verbose($"ArtifactProcessingManager.CollectArtifacts: Persist runsettings \n{runSettingsXml}");
+                _fileHelper.WriteAllTextToFile(Path.Combine(_testSessionProcessArtifactFolder, "runsettings.xml"), runSettingsXml);
+                var serializedExecutionComplete = _dataSerialized.SerializePayload(MessageType.ExecutionComplete, testRunCompleteEventArgs);
+                EqtTrace.Verbose($"ArtifactProcessingManager.CollectArtifacts: Persist ExecutionComplete message \n{serializedExecutionComplete}");
+                _fileHelper.WriteAllTextToFile(Path.Combine(_testSessionProcessArtifactFolder, "executionComplete.json"), serializedExecutionComplete);
+                EqtTrace.Verbose($"ArtifactProcessingManager.CollectArtifacts: Artifacts saved in {watch.Elapsed}");
+            }
+        }
+        catch (Exception e)
+        {
+            EqtTrace.Error("ArtifactProcessingManager.CollectArtifacts: Exception during artifact post processing: " + e);
+        }
+    }
+
+    public async Task PostProcessArtifacts()
+    {
+        if (!_featureFlag.IsEnabled(FeatureFlag.ARTIFACTS_POSTPROCESSING))
+        {
+            EqtTrace.Verbose("ArtifactProcessingManager.PostProcessArtifacts: Feature disabled");
+            return;
+        }
+
+        // This is not expected, anyway we prefer avoid exception for post processing
+        if (string.IsNullOrEmpty(_testSessionCorrelationId))
+        {
+            EqtTrace.Error("ArtifactProcessingManager.PostProcessArtifacts: Unexpected null testSessionCorrelationId");
+            return;
+        }
+
+        if (!_fileHelper.DirectoryExists(_processArtifactFolder))
+        {
+            EqtTrace.Verbose("ArtifactProcessingManager.PostProcessArtifacts: There are no artifacts to postprocess");
+            return;
+        }
+
+        var testArtifacts = LoadTestArtifacts();
+        if (testArtifacts?.Length > 0)
+        {
+            try
+            {
+                await MergeDataCollectorAttachments(testArtifacts);
+            }
+            finally
+            {
+                _fileHelper.DeleteDirectory(_processArtifactFolder, true);
```

Comment:
add `try/catch` we're in tmp folder if we're not able to cleanup....it's not a great issue